### ISSUE TITLE
Support merging of multiple cloud-init datasources

### ISSUE THREAD

**Issue by janeczku** [rancher/os#1536]
Currently when the `cloud_init.datasources` configuration array contains multiple sources (eg. multiple of `file` or `url`) only the first available source is ever applied by the cloud-init service.

The desired behavior is that multiple available datasources should be merged into the config. See also: http://cloudinit.readthedocs.io/en/latest/topics/merging.html

**Comment by SvenDowideit** [comment]
if there's a merge failure / url failure, then we're going to fail out.

**Comment by janeczku** [comment]
@SvenDowideit Need to sync on the merge behavior. E.g. multiple cloud-configs that are setting the same string key or array key.

**Comment by SvenDowideit** [comment]
@janeczku yup - for the short term, tell me what you want to have happen, and if I have time, I'll add configurability

**Comment by SvenDowideit** [comment]
I'm going to assume that we can have only one (successful) metadata source, and thus we're only merging `#cloud-config` and `#compose` sources.

**Comment by SvenDowideit** [comment]
bizzarely, if i may any comment in an issue i'm assigned to, but didn't create, GH unassigns me.
